jiant/tasks/lm.py

```diff
@@ -172,3 +175,102 @@ def __init__(self, path, *args, **kw):
             "val": os.path.join(path, "valid.sentences.txt"),
             "test": os.path.join(path, "test.sentences.txt"),
         }
+
+
+class MaskedLanguageModelingTask(LanguageModelingTask):
+    """
+       Generic Masked Language Modeling Task
+    """
+
+    pass
+
+
+@register_task("mlm", rel_path="WikiText103/")
+class MLMTask(MaskedLanguageModelingTask):
+    """
+    Masked language modeling task on Toronto Books dataset
+    Attributes:
+        max_seq_len: (int) maximum sequence length
+        min_seq_len: (int) minimum sequence length
+        files_by_split: (dict) files for three data split (train, val, test)
+    """
+
+    def __init__(self, path, *args, **kw):
+        super().__init__(path, *args, **kw)
+        self.files_by_split = {
+            "train": os.path.join(path, "train.sentences.txt"),
+            "val": os.path.join(path, "valid.sentences.txt"),
+            "test": os.path.join(path, "test.sentences.txt"),
+        }
+
+    def update_metrics(self, out, batch=None):
+        # self.scorer1(logits,labels)
+        self.scorer1(out["loss"].mean())
+        return
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text
+        Args:
+            path: (str) data file path
+        """
+        import csv
+
+        f = open(path, "r")
```

Comment:
I'm not very sure about this. When I worked on blimp, I remember I saw wikitext103 is already tokenized by Moses. If that is the case, you may need to detokenize it first before tokenize using self._tokenizer_name.